### PR TITLE
[Enhancement] Add favorite filter in Starter UI

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes?

- You can now register and cancel your favorite Pokémon on the starter selection UI.
- A Favorite Pokémon filter has also been added.

## Why am I doing these changes?

- Many users have requested this feature. (Including myself!)

## What did change?

- A favoriteIcon has been added to the Pokémon container.
- Allows you to add or remove Pokémon from your favorites using the newly added "Add to Favorites" and "Remove from Favorites" menu options.
- A favorite attribute has been added to the starter attribute. This ensures that the favorite status is saved and loaded even if you restart the game.

left issues
- [x] Check English translation for the added texts. ( checked by @Opaque02. thanks! )
- [x] Find a proper icon for Favorite. ( thanks, all art team! )

### Screenshots/Videos

https://github.com/user-attachments/assets/d268b086-efe9-4c42-a74c-ad7126edbf1f

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by returntoice — APPROVED** [review]
Korean translations LGTM

**Comment by frutescens** [comment]
https://github.com/user-attachments/assets/90c5c949-284c-4cd4-a25e-6638075b2656

image

**Comment by KimJeongSun** [comment]
@frutescens 
Thank you for your report!  The error has been fixed.

**Issue Cause**
The error occurred because `starterPreferences` was not defined at all for the specific species. (Previously, error handling was implemented to ensure no error arose when values for the "favorite" key were undefined, but in this case, it was unforeseen and occurred only for certain Pokémon.)

**Solution**
In other functionalities like shiny/nature, the same condition was handled by initializing `starterPreferences` to `{}` if it was undefined. I realized it would be better to consider this for newly added attributes as well.

Therefore, I added code to initialize `starterPreferences` to `{}` if it is undefined within the loop that checks the dex for all starters when `starterPreferences` is first loaded.

Thank you!

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -910,6 +916,9 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
       this.allSpecies.forEach((species, s) => {
         const icon = this.starterContainers[s].icon;
         const dexEntry = this.scene.gameData.dexData[species.speciesId];
+        if (!this.starterPreferences[species.speciesId]) {
+          this.starterPreferences[species.speciesId] = {};
+        }
```
Comment (torranx) [inline src/ui/starter-select-ui-handler.ts:921]:
```suggestion
        this.starterPreferences[species.speciesId] = this.starterPreferences[species.speciesId] ?? {};
```

Comment (KimJeongSun) [reply src/ui/starter-select-ui-handler.ts]: That's a good suggestion! There are no issues with the functionality after the update. Thank you!

**Review by torranx — APPROVED** [review]
looks good to me

**Pull request merged**